src/sdp/SDP.ts

```diff
@@ -375,15 +377,22 @@ export default class SDP {
    * negotiate with the back end to determine whether to use layers allocation header extension
    * this will not add the packet overhead unless negotiated to avoid waste
    */
-  withVideoLayersAllocationRtpHeaderExtension(): SDP {
-    const sections = SDP.splitSections(this.sdp);
+  withVideoLayersAllocationRtpHeaderExtension(previousSdp: SDP): SDP {
+    const url = `http://www.webrtc.org/experiments/rtp-hdrext/video-layers-allocation00`
 
+    // According to https://webrtc.googlesource.com/src/+/b62ee8ce94e5f10e0a94d6f112e715cc4d0cd9dc,
+    // RTP header extension ID change would result in a hard failure. Therefore if the extension exists
+    // in the previous SDP, use the same extension ID to avoid the failure. Otherwise use a new ID
+    let previousId = previousSdp ? previousSdp.getRtpHeaderExtensionId(url) : -1;
+    const id = previousId === -1 ? this.getUniqueRtpHeaderExtensionId(SDP.splitLines(this.sdp)) : previousId;
+
+    const sections = SDP.splitSections(this.sdp);
     const newSections = [];
     for (let section of sections) {
-      if (/^m=video/.test(section)) {
+      if (/^m=video/.test(section) && SDP.getRtpHeaderExtensionIdInSection(section, url) === -1) {
+        // Add RTP header extension only when it does not already exist
```

Comment:
To be safe, should we check to see if its different and override it if so?

Reply:
In case that there is a different extension ID than previous SDP, the problem is a lot bigger since there could be multiple changes in the initial SDP. In that case, simply overriding the extension ID may not solve the problem since the id in previous SDP might be used by some other extensions. My understanding is that we should probably treat the initial SDP's extension ID as ground truth here since if the initial SDP is incorrect, it does not seem fixable in this function.

Reply:
So this is actually the issue that we previously ran into that caused all the issues. When the header extension showed up again, it had a different ID (yes this is a browser bug, but since SDP munging is sorta in the 'technically works but not official' region of support I just want to be careful :) ). What about just sanity checking and removing the line if it exists?

Reply:
In the new PR, I added https://github.com/aws/amazon-chime-sdk-js/pull/2195/files#diff-d6a73856184ac9b6920494e5e8fcc8be715f515d30c8871cc9a8891ebc7c87fdR411 to override the extension ID if changed, it also removes the extension if the ID changed and the old ID is used by some other extensions